### PR TITLE
[vm] Create an entry point for initializing LibraVM on chain configur…

### PR DESCRIPTION
**Pull request by runtian-zhou**
…ations

## Motivation

Provide a public api for initializing all the on chain configs required by the LibraVM. We will use this API to setup the initialization phase of LibraVM in VMValidator and Executor in the future.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

File: language/vm/vm-runtime/src/data_cache.rs [hunk 1]
```diff
@@ -80,6 +80,22 @@ impl<'block> RemoteCache for BlockDataCache<'block> {
     }
 }
 
+pub(crate) struct RemoteStorage<'a>(&'a dyn StateView);
```
Comment (runtian-zhou) [inline language/vm/vm-runtime/src/data_cache.rs]: @sunshowers is there a better way to do this other than creating a wrapper struct?

Comment (sunshowers) [reply language/vm/vm-runtime/src/data_cache.rs]: Can you do `impl<'a> RemoteCache for &'a dyn StateView`?

Looks good otherwise!

Comment (runtian-zhou) [reply language/vm/vm-runtime/src/data_cache.rs]: `   = note: expected type `&dyn data_cache::RemoteCache`
              found type `&dyn libra_state_view::StateView``
I got some error messages like this :(

Comment (sunshowers) [reply language/vm/vm-runtime/src/data_cache.rs]: ah ok, want to put in a comment about this?

**Pull request closed without merging**